src/components/Form/FileUpload/FileUpload.test.tsx

```diff
@@ -347,6 +387,59 @@ describe("upload action", () => {
     expect(onDrop).toHaveBeenCalledTimes(0);
   });
 
+  it("show error state and error message on drop of invalid extension file", async () => {
+    const onDrop = jest.fn();
+    const { container } = createComponent(
+      defaultParams => ({
+        ...defaultParams,
+        onDrop,
+        fileList: []
+      }),
+      "file-upload-6"
+    );
+
+    const file = new File([""], "test.png", {
+      type: "image/png"
+    });
+
+    const eventData = {
+      dataTransfer: {
+        files: [file]
+      }
+    };
+
+    const dropZone = container.querySelector(".upload-button-wrapper") as Element;
+    const dropEvent = createEvent.drop(dropZone, eventData);
+
+    await waitFor(() => {
+      fireEvent(dropZone, dropEvent);
+    });
+
+    const dragDropWrapper = container.querySelector(".file-dropzone") as Element;
+    const errorMessage = container.querySelector(".file-selector-sub-text") as Element;
+
+    expect(dragDropWrapper.classList.contains("error")).toBeTruthy();
+    expect(errorMessage.classList.contains("error")).toBeTruthy();
+    expect(errorMessage.innerHTML).toStrictEqual(
+      `Invalid file format. Supported formats are: ${defaultParams.accept}.`
+    );
+  });
+
+  it("sub text or custom text is visible", async () => {
+    const { container } = createComponent(
+      defaultParams => ({
+        ...defaultParams,
+        fileList: [],
+        subText: "This is sub heading."
+      }),
+      "file-upload-6"
+    );
+    const subTextElement = container.querySelector(".file-selector-sub-text") as Element;
+
+    expect(subTextElement).not.toBeNull();
```

Comment:
```suggestion
    expect(subTextElement).toBeVisible();
```

this way we make sure that the element is actually on the page, not hidden, not unavailable, but visible by the user